HACKING.md

```diff
@@ -0,0 +1,160 @@
+# Hacking on snapd
+
+Hacking on snapd is fun and straightfoward. The code is extensively
+unit tested and we use the [spread](https://github.com/snapcore/spread)
+integration test framework for the integration/system level tests.
+
+## Development
+
+### Setting up a GOPATH
+
+When working with the source of Go programs, you should define a path within
+your home directory (or other workspace) which will be your `GOPATH`. `GOPATH`
+is similar to Java's `CLASSPATH` or Python's `~/.local`. `GOPATH` is documented
+[http://golang.org/pkg/go/build/](online) and inside the go tool itself
```

Comment:
I know you're just moving stuff around, but while you're here: that link isn't. I believe the intent was to have it the other way around, ie a link that said "online" and pointed you to that URL. It needs reversing: `[online](http://golang.org/pkg/go/build/)`.